packages/api/lambdas/sf-event-sqs-to-db-records/write-pdr.js

```diff
@@ -2,33 +2,175 @@
 
 const {
   tableNames,
+  getRecordCumulusId,
 } = require('@cumulus/db');
 const {
   getMessagePdrName,
   messageHasPdr,
+  getMessagePdrStats,
+  getMessagePdrPANSent,
+  getMessagePdrPANMessage,
+  getPdrPercentCompletion,
 } = require('@cumulus/message/PDRs');
 const {
   getMetaStatus,
+  getMessageWorkflowStartTime,
+  getWorklowDuration,
 } = require('@cumulus/message/workflows');
 
 const Pdr = require('../../models/pdrs');
 
+const {
+  getCumulusIdFromRawInsertQueryResult,
+} = require('./utils');
+
+const generatePdrRecord = ({
+  cumulusMessage,
+  collectionCumulusId,
+  providerCumulusId,
+  executionCumulusId,
+  now = Date.now(),
+}) => {
+  const stats = getMessagePdrStats(cumulusMessage);
+  const progress = getPdrPercentCompletion(stats);
+  const timestamp = now;
+  const workflowStartTime = getMessageWorkflowStartTime(cumulusMessage);
+  return {
+    name: getMessagePdrName(cumulusMessage),
+    status: getMetaStatus(cumulusMessage),
+    pan_sent: getMessagePdrPANSent(cumulusMessage),
+    pan_message: getMessagePdrPANMessage(cumulusMessage),
```

Comment:
What is `pan`?

Reply:
To be honest, I've never known what it means. It seems like these NASA docs explain it though

https://earthdata.nasa.gov/esdis/eso/standards-and-references/polling-with-delivery-record-pdr-mechanism
https://cdn.earthdata.nasa.gov/conduit/upload/6376/ESDS-RFC-030v1.0.pdf